src/lib/core/data-table/data-table.ts

```diff
@@ -92,9 +98,13 @@ export class CdkTable implements CollectionViewer {
   /** Set of templates that used as the data row containers. */
   @ContentChildren(CdkRowDef) _rowDefinitions: QueryList<CdkRowDef>;
 
-  constructor(private _changeDetectorRef: ChangeDetectorRef) {
+  constructor(private _differs: IterableDiffers,
+              private _changeDetectorRef: ChangeDetectorRef) {
     console.warn('The data table is still in active development ' +
         'and should be considered unstable.');
+
+    // TODO(andrewseguin): Add trackby function input.
+    this._dataDiffer = this._differs.find([]).create();
```

Comment:
I don't quite follow how this works; why is this `find` necessary? Can you not just create an `IterableDiffer` directly? Why is the DI system involved? 
(probably just needs explanation in comments)